src/pages/products/ProductItems.jsx

```diff
@@ -0,0 +1,20 @@
+import styles from "./Products.module.css";
+import React from "react";
+import { useProducts } from "../../context/product-context";
+import ProductCard from "../../components/ProductCard";
+
+export default function ProductItems() {
+  const { finalProducts } = useProducts();
+
+  return (
+    <section
+      className={`min-h-100 grid ${styles.grid__cols__auto} ${styles.grid__col__1} ${styles.grid__cols__2} gap-2 my-3 px-md`}>
+      {finalProducts.length === 0 && (
+        <div className="h1">No Products Available :(</div>
+      )}
+      {finalProducts.map((product) => (
+        <ProductCard product={product}/>
```

Comment:
give a key when you are maping.

Reply:
Had passed the key prop to the component instead of here where its being mapped. Good catch!